PORTAL/jobs/_utils.py

```diff
@@ -842,7 +959,12 @@ def run_bg(argv, logfile=None, *, cwd=None, env=None):
 ##################################
 # file utils
 
-def check_shell_str(value, *, required=True, allowspaces=False):
+def check_shell_str(
+        value: Any,
```

Comment:
resisting

Reply:
This is a `Union[GitRef, str, Dict[str, str], Tuple[str, str, str, str, str, str]]` ;)